components/blocks/Recipe.template.tsx

```diff
@@ -0,0 +1,115 @@
+import { wrapFieldsWithMeta } from 'tinacms';
+import React, { useEffect, useRef, useState } from 'react';
+import debounce from 'lodash/debounce';
+
+export const RecipeBlock = {
+  name: 'recipeBlock',
+  label: 'Recipe Block',
+  fields: [
+    {
+      name: 'title',
+      label: 'Heading Title',
+      type: 'string',
+    },
+    {
+      name: 'description',
+      label: 'Description',
+      type: 'string',
+    },
+    {
+      type: 'string',
+      name: 'code',
+      label: 'Code',
+      ui: {
+        component: wrapFieldsWithMeta(({ field, input, meta }) => {
```

Comment:
would be nice if this component was moved to its own file to clean up the template

Reply:
done